simple.go

```diff
@@ -58,7 +62,7 @@ func (c *SimpleCache) set(key, value interface{}) (interface{}, error) {
 		item.value = value
 	} else {
 		// Verify size not exceeded
-		if len(c.items) >= c.size {
+		if (len(c.items) >= c.size) && c.size != -1 {
```

Comment:
@aaronwinter 
ref: https://github.com/bluele/gcache/pull/38/files#r127405430

Could you change to `if c.size > 0 && len(c.items) >= c.size` ?